src/main/java/io/vertx/kafka/client/consumer/KafkaReadStream.java

```diff
@@ -69,6 +69,18 @@
   @Override
   KafkaReadStream<K, V> endHandler(@Nullable Handler<Void> endHandler);
 
+  /**
+   * Returns the current fetch amount.
+   *
+   * <ul>
+   *   <i>If the stream is in <i>flowing</i> mode will return {@link Long#MAX_VALUE}.</i>
+   *   <li>If the stream is in <i>fetch</i> mode, will return the current number of elements still to be delivered or 0 if paused.</li>
+   * </ul>
+   *
+   * @return current fetch amount
+   */
+  long currentFetchAmount();
```

Comment:
Another approach would be to return `OptionalLong` since `Long.MAX_VALUE` for *flowing* mode is an implementation detail. In this case we can return `empty` if it's in *flowing* mode.

Reply:
rather return rather Long , optional will not work well with Vert.x code generator

Reply:
that being said doing fetch(Long.MAX_VALUE) and resume() should work the same, so returning Long.MAX_VALUE makes sense as your code does not need if/else statements

Reply:
Ok